src/nacl/encoding.py

```diff
@@ -11,73 +11,92 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-
-
 import base64
 import binascii
-from typing import SupportsBytes
+from abc import ABCMeta, abstractmethod
+from typing import SupportsBytes, Type
+
+
+# TODO: when the minimum supported version of Python is 3.8, we can import
+# Protocol from typing, and replace Encoder with a Protocol instead.
+class _Encoder(metaclass=ABCMeta):
+    @staticmethod
+    @abstractmethod
+    def encode(data: bytes) -> bytes:
+        pass
```

Comment:
Coverage is upset about this, but we can entirely replace `pass` in these cases with a comment and it will be happy:

```python
@staticmethod
@abstractmethod
def encode(data: bytes) -> bytes:
    """
    The encode method
    """
```

Or some possibly less useless comment 😄